core/database/src/main/java/com/mifos/room/entities/templates/loans/LoanTemplate.kt

```diff
@@ -44,115 +44,115 @@ import kotlinx.parcelize.Parcelize
  */
 @Parcelize
 data class LoanTemplate(
-    var clientId: Int? = null,
+    val clientId: Int? = null,
 
-    var clientAccountNo: String? = null,
+    val clientAccountNo: String? = null,
 
-    var clientName: String? = null,
+    val clientName: String? = null,
 
-    var clientOfficeId: Int? = null,
+    val clientOfficeId: Int? = null,
 
-    var loanProductId: Int? = null,
+    val loanProductId: Int? = null,
 
-    var loanProductName: String? = null,
+    val loanProductName: String? = null,
 
-    var isLoanProductLinkedToFloatingRate: Boolean? = null,
+    val isLoanProductLinkedToFloatingRate: Boolean? = null,
 
-    var fundId: Int? = null,
+    val fundId: Int? = null,
 
-    var fundName: String? = null,
+    val fundName: String? = null,
 
-    var currency: Currency? = null,
+    val currency: Currency? = null,
 
-    var principal: Double? = null,
+    val principal: Double? = null,
 
-    var approvedPrincipal: Double? = null,
+    val approvedPrincipal: Double? = null,
 
-    var proposedPrincipal: Double? = null,
+    val proposedPrincipal: Double? = null,
 
-    var termFrequency: Int? = null,
+    val termFrequency: Int? = null,
 
-    var termPeriodFrequencyType: TermPeriodFrequencyType? = null,
+    val termPeriodFrequencyType: TermPeriodFrequencyType? = null,
 
-    var numberOfRepayments: Int? = null,
+    val numberOfRepayments: Int? = null,
 
-    var repaymentEvery: Int? = null,
+    val repaymentEvery: Int? = null,
 
-    var repaymentFrequencyType: RepaymentFrequencyType? = null,
+    val repaymentFrequencyType: RepaymentFrequencyType? = null,
 
-    var interestRatePerPeriod: Double? = null,
+    val interestRatePerPeriod: Double? = null,
 
-    var interestRateFrequencyType: InterestRateFrequencyType? = null,
+    val interestRateFrequencyType: InterestRateFrequencyType? = null,
 
-    var annualInterestRate: Double? = null,
+    val annualInterestRate: Double? = null,
 
-    var isFloatingInterestRate: Boolean? = null,
+    val isFloatingInterestRate: Boolean? = null,
 
-    var amortizationType: AmortizationType? = null,
+    val amortizationType: AmortizationType? = null,
 
-    var interestType: InterestType? = null,
+    val interestType: InterestType? = null,
 
-    var interestCalculationPeriodType: InterestCalculationPeriodType? = null,
+    val interestCalculationPeriodType: InterestCalculationPeriodType? = null,
 
-    var allowPartialPeriodInterestCalcualtion: Boolean? = null,
+    val allowPartialPeriodInterestCalcualtion: Boolean? = null,
 
-    var transactionProcessingStrategyId: Int? = null,
+    val transactionProcessingStrategyId: Int? = null,
 
-    var graceOnArrearsAgeing: Int? = null,
+    val graceOnArrearsAgeing: Int? = null,
 
-    var timeline: Timeline? = null,
+    val timeline: Timeline? = null,
 
-    var productOptions: List<ProductOptions> = ArrayList(),
+    val productOptions: List<ProductOptions> = ArrayList(),
```

Comment:
@itsPronay Please use emptyList(), emptySet(), or emptyMap() instead of ArrayList(), HashSet(), or other explicit collection initializations in the database module.
By minimizing pointless object creation, enforcing immutability to prevent accidental changes, and adhering to Kotlin best practices for cleaner, more maintainable code, using these functions increases efficiency.